openHarmony/openHarmony_node.js

```diff
@@ -3321,11 +3321,15 @@ $.oGroupNode.prototype.importImageAsTVG = function(path, alignment, nodePosition
  *
  * @returns {$.oDrawingNode} the created node
  */
-$.oGroupNode.prototype.importImageSequence = function(imagesPath, exposureLength, convertToTvg, alignment, nodePosition) {
+$.oGroupNode.prototype.importImageSequence = function(imagesPath, exposureLength, convertToTvg, alignment, nodePosition, extendScene) {
   if (typeof exposureLength === 'undefined') var exposureLength = 1;
   if (typeof alignment === 'undefined') var alignment = "ASIS"; // create an enum for alignments?
   if (typeof nodePosition === 'undefined') var nodePosition = new this.$.oPoint(0,0,0);
-
+  if (typeof extendScene === 'undefined') var extendScene = false;
+  
+  // Extend the scene?
+  if (extendScene && this.scene.length < imagesPath.length * exposureLength) this.scene.length = imagesPath.length * exposureLength;
```

Comment:
I think this is where the off by one error occurs: after checking, it seems (because Harmony frames start at 1) that `this.scene.length` corresponds to the last visible frame number. Which doesn't match with `array.length` which is always one more than the final index.

The number parsing is only used when exposureLength == 0. In this case, numbers on the images will be used to identify the frame at which they appear. (basically it's a regex that will match any numeral digits visible at the end of a file name separated by whichever character is not a digit just before) 

We should find a way to handle that use case too?
For example by parsing the names first to identify the greatest visible number? 
Since extendScene is false by default, maybe we can do that in a separate change. It's a shame that extendScene can't be at the same position as for importQt, but it's my fault really 😛 

Let me know if these informations help you figure out the off by one thing.

Reply:
I've made a refactor to handle all cases: https://github.com/cfourney/OpenHarmony/commit/56bae6315a3a43d903b4e986e592775fd2b32b04